### ISSUE TITLE
"UndefinedConversionError" occurs with multibyte char data in v2.0.0.alpha

### ISSUE THREAD

**Issue by darklore** [awslabs/aws-fluent-plugin-kinesis#118]
"Encoding::UndefinedConversionError" occurs In fluent-plugin-kinesis version 2.0.0.alpha.1.1 when using multibyte character data as input.
Without multibyte character, no error occurred.

Here is td-agent.log when the error occurred.

/var/log/td-agent/td-agent.log
```
2017-03-30 13:45:57 +0000 [warn]: #0 failed to flush the buffer. retry_time=0 next_retry_seconds=2017-03-30 13:45:58 +0000 chunk="54bf2e98e70eecdac64684370223aa66" error_class=Encoding::UndefinedConversionError error="U+3042 from UTF-8 to ASCII-8BIT"
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/aggregator.rb:51:in `initialize'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/aggregator.rb:51:in `new'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/aggregator.rb:51:in `block in aggregate'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/aggregator.rb:50:in `map'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/aggregator.rb:50:in `aggregate'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/out_kinesis_streams_aggregated.rb:51:in `block in write'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/api.rb:90:in `batch_request_with_retry'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis.rb:107:in `block in write_records_batch'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis_helper/api.rb:85:in `split_to_batches'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/kinesis.rb:105:in `write_records_batch'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kinesis-2.0.0.alpha.1.1/lib/fluent/plugin/out_kinesis_streams_aggregated.rb:44:in `write'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.14/lib/fluent/compat/output.rb:131:in `write'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.14/lib/fluent/plugin/output.rb:1038:in `try_flush'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.14/lib/fluent/plugin/output.rb:1263:in `flush_thread_run'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.14/lib/fluent/plugin/output.rb:415:in `block (2 levels) in start'
  2017-03-30 13:45:57 +0000 [warn]: #0 /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.14/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
```

Here is a reproduction procedure.

1. Install td-agent and fluent-plugin-kinesis 2.0.0.alpha.

``` 
curl -L https://toolbelt.treasuredata.com/sh/install-ubuntu-trusty-td-agent2.sh | sh
sudo td-agent-gem install fluent-plugin-kinesis -v 2.0.0.alpha.1.1
sudo td-agent-gem clean
```

2. Edit td-agent.conf to use out_kinesis_streams_aggregated as follows.

/etc/td-agent/td-agent.conf
```
<source>
  @type tail
  format json
  path /tmp/kinesis.log
  # pos_file /tmp/kinesis.log.pos
  read_from_head true

  tag kinesis
</source>

<match kinesis>
  @type kinesis_streams_aggregated
  stream_name test-stream
  region ap-northeast-1

  time_format %s
  include_time_key true

  buffer_type file
  buffer_path /var/log/td-agent/buffer/kinesis
  # buffer_chunk_limit 500k
  # buffer_queue_limit 1
  buffer_chunk_limit 500k
  buffer_queue_limit 16384

  flush_interval 1s
  try_flush_interval 0.1s
  queued_chunk_flush_interval 0.01s
  # num_threads 4
</match>

<match fluent.**>
  @type stdout
</match>

<match **>
  @type stdout
</match>
```

3. Set input data includes multibyte character

/tmp/kinesis.log
```
{"memo":"あいうえお"}
{"memo":"あいうえお"}
```

4. Restert td-agent
```
sudo service td-agent restart
```

**Comment by riywo** [comment]
Hi @darklore ,

Thank you for your reporting. I could confirm this is caused with kinesis_streams_aggregated. I'll look into this issue.

Best,

**Comment by riywo** [comment]
It should be fixed in 2.0.0.alpha.1.2 

cc: #112
